Add some unittests for util and sysprep

artmello opened the pull request.

artmello committed:
Add some unittests for util and sysprep

bocekm commented on tox.ini:
```diff
@@ -0,0 +1,13 @@
+
+[testenv]
+deps =
+    -rrequirements-tests.txt
```
The file you created is actually `requirements-test.txt`.

bocekm commented on tox.ini:
```diff
@@ -0,0 +1,13 @@
+deps =
+    -rrequirements-tests.txt
+commands =
+    py.test --flake8
```
I ran the tox by `make test-all` and I got an error:
```
ImportError while importing test module '/home/mbocek/repos/rhel_pkgs/redhat-upgrade-tool-github/tests/test_sysprep.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_sysprep.py:2: in <module>
    from mock import MagicMock
E   ImportError: No module named mock
```
- Ok, I went ahead and added mock to the `requirements-test.txt` but to no avail, the same error. It seems like pip doesn't install the packages mentioned in the file.
- So I added `sitepackages = True` to the `tox.ini` because I have mock on my machine installed. I didn't have `pytest-flake8` installed though, so I got another error. So I installed pytest-flake8 on my machine - btw, it's missing in the `requirements-test.txt` but since the file seems to not have any effect on the virtualenv, it doesn't matter.
- Then, I run `make test-all` and it gets stuck on the following and it just does not exit.
   ```
   ==================================================================================== test session starts =====================================================================================
   platform linux2 -- Python 2.7.13, pytest-3.2.1, py-1.4.34, pluggy-0.4.0
   rootdir: /home/mbocek/repos/rhel_pkgs/redhat-upgrade-tool-github, inifile: tox.ini
   plugins: flake8-0.8.1
   collected 27 items                                                                                                                                                                            
   
   redhat-upgrade-tool.py
   ```
- I haven't used `pytest` before, I use `nosetests` instead, so I tried to replace the `py.test --flake8` tox command with `nosetests -v --rednose --nocapture` and it finally finished well. But hey, it was a fight :-D Even though I get the concept of tox I haven't used it before, so maybe I'm missing some step.

artmello committed:
Fixed tests execution

- Fixed tox.ini to call requirementes-test.txt correctly
- Moved from pytest to nose

artmello commented:
@bocekm thx for checking this out. I pushed some changes that, hopefully, will fix running tests. Let me know how that works for you
